imports/plugins/core/components/lib/hoc.js

```diff
@@ -26,6 +32,56 @@ export function withCurrentUser(component) {
 }
 
 
+/**
+ * @name withMoment
+ * @method
+ * @summary A wrapper to reactively inject the moment package into a component
+ * @param {Function|React.Component} component - the component to wrap
+ * @return {Function} the new wrapped component with a "moment" prop
+ * @memberof Components
+ */
+export function withMoment(component) {
+  return lifecycle({
+    componentDidMount() {
+      try {
+        import("moment").then(moment => {
+          this.setState({
+            moment: moment.default
+          });
+        });
+      } catch (error) {
+        Logger.debug(error, "moment.js async import error");
+      }
```

Comment:
This `catch` will not catch errors because they will get swallowed by the promise. You should catch the promise instead:

```js
import("moment")
  .then(...)
  .catch((error) => { Logger.debug(error, "moment.js async import error"); });
```

Or you can `await` the import line and then `try/catch` will work.

Reply:
Which is better practice / should we be using? Promieses or async / await?

Reply:
In many cases I'd say async / await, but something like this... I'm not sure that adding `async` to a React instance method would be a good idea. Maybe it would work, but I'd lean toward the safety of promise+then.